ramips: add support for D-Link DIR-2055 A1

kewiha opened the pull request:
Add support for D-Link DIR-2055 A1 based on similarities to DIR-1960 A1, as well as various DIR-8xx A1 models. Existing DIR-1960 A1 openwrt "factory" firmware installs without modifications via the D-Link Recovery GUI and has no known incompatibilities with the DIR-2055 A1.

Changes to be committed:
new file:   target/linux/ramips/dts/mt7621_dlink_dir-2055-a1.dts
modified:   target/linux/ramips/image/mt7621.mk
modified:   target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
modified:   target/linux/ramips/mt7621/base-files/lib/upgrade/platform.sh

Specifications:

    Board: Not known
    SoC: MediaTek MT7621 Family (MT7621AT)
    RAM: 256 MB (Micron 9OK17 D9PTK, should be DDR3 MT41K128M16JT-125)
    Flash: 128 MB (Winbond W29N01HVSINA)
    WiFi: MediaTek MT7615 Family (MT7615N x2)
    Switch: 1 WAN, 4 LAN (Gigabit)
    Ports: 1 USB 3.0 (front)
    Buttons: Reset, WiFi Toggle, WPS
    LEDs: Power (white/orange), Internet (white/orange), WiFi 2.4G (white), WiFi 5G (white)

Notes:

    Only known difference vs. the DIR-1960 A1 is that the DIR-2055 A1 doesn't have a USB activity LED

Serial port:

    Tested to be identical to various DIR-8xx A1 models with a similar enclosure/pcb design:
        Parameters: 57600, 8N1, 3.3V TTL no flow control
        Location: J1 header (close to the Reset, WiFi and WPS buttons)
        Pinout: 1 - VCC 2 - RXD 3 - TXD 4 - GND
            Did not connect VCC when using

Installation:

    D-Link Recovery GUI: power down the router, press and hold the reset button, then re-plug it. Keep the reset button pressed until the power LED starts flashing orange, manually assign a static IP address under the 192.168.0.xxx subnet (e.g. 192.168.0.2) and go to http://192.168.0.1

    Some modern browsers may have problems flashing via the Recovery GUI, if that occurs consider uploading the firmware through cURL:

    curl -v -i -F "firmware=@file.bin" 192.168.0.1

Signed-off-by: Keith Harrison <keithh@protonmail.com>

raenye commented:
Why not add as `DEVICE_ALT`? (see, e.g., #2924)

Lucky1openwrt commented:
> Why not add as `DEVICE_ALT`? (see, e.g., #2924)

how's the dlink u-boot know about `DEVICE_ALT` ? did you tell them ?
how's that help filling the correct header info into the firmware so it will pass it's checks "model number in this case"

raenye commented:
> how's the dlink u-boot know about `DEVICE_ALT` ? did you tell them ?

OP said:
> Existing DIR-1960 A1 openwrt "factory" firmware installs without modifications via the D-Link Recovery GUI

Lucky1openwrt commented:
> OP said:
> 
> > Existing DIR-1960 A1 openwrt "factory" firmware installs without modifications via the D-Link Recovery GUI

MY dir-1960-a1 doesn't look
but that 1st DIR-2660-A1 didn't but then next batch did and we had to add it
so if you know there are no newer batches or won't ever be then ok
but better to be safe then sorry

Lucky1openwrt commented:
@kewiha 
how are you going with this ?

Lucky1openwrt commented:
Hi @kewiha i'm just another Contributor like you
just some changes I think look wrong to me "tho I may be wrong"

you don't need the include line in the DTS as its inside the linked DTSI
and yes you could save a few bytes if you remove kmod-usb-ledtrig-usbport 
as you don't need it not having an led for USB

I can only give you a review tho I can't merged it

Lucky1openwrt commented:
FYI the include file in the DTS has changed name
from `#include "mt7621_dlink_dir-xx60-a1.dtsi"`
to `#include "mt7621_dlink_dir_nand_128m.dtsi"`

namiltd commented:
After [8d7fa0918143cf09049d4631a0cdef7962554147 ](https://github.com/openwrt/openwrt/commit/8d7fa0918143cf09049d4631a0cdef7962554147) and https://github.com/openwrt/openwrt/commit/d646ea8a00aab7998e4d7bf80ba256cd65347d2c Your PR needs a rebuild

Lucky1openwrt commented:
@kewiha 
this is what I think it would end up like 
with blue led's & the added wifi button
https://github.com/Lucky1openwrt/openwrt/commit/417a8008f2f0ec004b976a42a5a51fac0e23ccb1

DragonBluep commented:
Please rebase this branch and reorganize your commit message. Each line should not exceed 75 characters. And add `< >` to your email address.

kewiha committed:
ramips: add support for D-Link DIR-2055 A1

Add support for D-Link DIR-2055 A1 based on similarities to DIR-1960 A1,
as well as various DIR-8xx A1 models. Existing DIR-1960 A1 openwrt
"factory" firmware installs without modifications via the D-Link Recovery
GUI and has no known incompatibilities with the DIR-2055 A1.

Changes to be committed:
new file:   target/linux/ramips/dts/mt7621_dlink_dir-2055-a1.dts
modified:   target/linux/ramips/image/mt7621.mk
modified:   target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
modified:   target/linux/ramips/mt7621/base-files/lib/upgrade/platform.sh

Specifications:

    Board: Not known
    SoC: MediaTek MT7621 Family (MT7621AT)
    RAM: 256 MB (Micron 9OK17 D9PTK, should be DDR3 MT41K128M16JT-125)
    Flash: 128 MB (Winbond W29N01HVSINA)
    WiFi: MediaTek MT7615 Family (MT7615N x2)
    Switch: 1 WAN, 4 LAN (Gigabit)
    Ports: 1 USB 3.0 (front)
    Buttons: Reset, WiFi Toggle, WPS
    LEDs: Power (white/orange), Internet (white/orange),
          WiFi 2.4G (white), WiFi 5G (white)

Notes:

    Only known difference vs. the DIR-1960 A1 is that the DIR-2055 A1
    doesn't have a USB activity LED

Serial port:

    Tested to be identical to various DIR-8xx A1 models with a similar
    enclosure/pcb design:
        Parameters: 57600, 8N1, 3.3V TTL no flow control
        Location: J1 header (close to the Reset, WiFi and WPS buttons)
        Pinout: 1 - VCC 2 - RXD 3 - TXD 4 - GND
            Did not connect VCC when using

Installation:

    D-Link Recovery GUI: power down the router, press and hold the reset
    button, then re-plug it. Keep the reset button pressed until the power
    LED starts flashing orange, manually assign a static IP address under
    the 192.168.0.xxx subnet (e.g. 192.168.0.2) and go to
    http://192.168.0.1

    Some modern browsers may have problems flashing via the Recovery GUI,
    if that occurs consider uploading the firmware through cURL:

    curl -v -i -F "firmware=@file.bin" 192.168.0.1

Signed-off-by: Keith Harrison <keithh@protonmail.com>

Lucky1openwrt commented:
just to let you know the code for the NOR flash 8xx routers for the radio on/off is located 
in the DTS Include file https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/dts/mt7621_dlink_dir-8xx.dtsi
but most if not all have the button installed

but the DIR-2055 is the 1st time I have seen the button loaded in the PCB for a NAND flash device
even tho the DIR-882-A1 and DIR-1960-A1 are the same PCB they Don't load the physical button
you would need to drill a hole in the case to add this as well to fit one
